pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -185,6 +202,88 @@ func (pt PipelineTask) ValidateName() *apis.FieldError {
 	return nil
 }
 
+// Validate custom task specifications - checking kind and fail if not yet supported features specified
+func (pt PipelineTask) ValidateCustomTask() (errs *apis.FieldError) {
+	errs = errs.Also(pt.validate())
+	if pt.TaskRef.Kind == "" {
+		errs = errs.Also(apis.ErrInvalidValue("custom task ref must specify kind", "taskRef.kind"))
+	}
+	// Conditions are deprecated so the effort to support them with custom tasks is not justified.
+	// When expressions should be used instead.
+	if len(pt.Conditions) > 0 {
+		errs = errs.Also(apis.ErrInvalidValue("custom tasks do not support conditions - use when expressions instead", "conditions"))
+	}
+	// TODO(#3133): Support these features if possible.
+	if pt.Retries > 0 {
+		errs = errs.Also(apis.ErrInvalidValue("custom tasks do not support retries", "retries"))
+	}
+	if pt.Resources != nil {
+		errs = errs.Also(apis.ErrInvalidValue("custom tasks do not support PipelineResources", "resources"))
+	}
+	if pt.Timeout != nil {
+		errs = errs.Also(apis.ErrInvalidValue("custom tasks do not support timeout", "timeout"))
+	}
+	return errs
+}
+
+// Validate bundle specifications - checking name and bundle
+func (pt PipelineTask) ValidateBundle() (errs *apis.FieldError) {
+	errs = errs.Also(pt.validate())
```

Comment:
We will call this twice (in a normal `Validate`), any reason ?

Reply:
I have deleted it from here and moved this in the `pipelineTask.Validate` so that it's easier to read.

Nope this was not called twice, `ValidateBundle` is only called when a bundle is getting validated.